e2e/v3/cluster3/cluster3.go

```diff
@@ -149,11 +152,24 @@ func (c *Cluster) wait() {
 		err := <-errCh
 		must.NoError(c.t, err)
 	}
+
+	// t.Skip() should not be run in a goroutine, so this check is separate,
+	// and by the time the above have passed, retries should not be necessary.
+	if c.enterprise {
+		_, _, err := c.nomadClient.Operator().LicenseGet(nil)
+		if err != nil && err.Error() == "Nomad Enterprise only endpoint" {
+			c.t.Skip("not an enterprise cluster")
+		} else {
+			must.NoError(c.t, err, must.Sprint("expect running Enterprise cluster"))
```

Comment:
doesn't the `else` need to be `else if err != nil` ? (i.e. this _is_ an enterprise cluster and we should allow the test to run)

Reply:
the `must.NoError()` covers nil-checking in that case.  if it's nil, then continue, else t.Fatal (via must)

example playground: https://go.dev/play/p/VriAyDQ3750

but if the syntax is confusing, I'm happy to tweak it to clarify intent.

Reply:
oh right, i just can't read